### PR TITLE
Hive: exit if we enter ProvisionFailed status

### PR DESCRIPTION
**Pull request by darthhexx**
### Which issue this PR addresses:

Fixes https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/15728844

### What this PR does / why we need it:

Currently the Hive manager is only checking for `spec.Installed`, but this does not check for any failure conditions and results in all failures taking 60 minutes to return. This PR adds a single failure condition, Provision Failed, but scaffolding has been put in place making it trivial to add any additional error conditions in the future.

### Test plan for issue:

Tested cluster installation failure propagation in a dedicated RP instance.

### Is there any documentation that needs to be updated for this PR?

No.

### CODE REVIEW

File: pkg/hive/manager.go [hunk 1]
```diff
@@ -160,10 +161,26 @@ func (hr *clusterManager) IsClusterDeploymentReady(ctx context.Context, doc *api
...
-	return false, err
+
+	for _, cond := range cd.Status.Conditions {
+		conditionStatus, found := checkFailureConditions[cond.Type]
+		if found && conditionStatus == cond.Status {
+			return false, fmt.Errorf("clusterdeployment has failed: %s == %s", cond.Type, cond.Status)
+		}
+	}
+
+	return false, nil
```
Comment (s-amann) [inline pkg/hive/manager.go:183]: wdyt about adding manager tests? I don't think we have any, and the logic here is growing

**Review by s-amann — APPROVED** [review]
lgtm, we should have a ticket for enhancing the tests if we are in a rush to get this out

File: pkg/hive/manager.go (hunk 1 above)
Comment (darthhexx) [reply pkg/hive/manager.go:183]: Good point. I was racing to get this across the line, but there actually is time. I've added some tests.

**Review by s-amann — APPROVED** [review]
lgtm, thank you for adding the tests

**Review by cadenmarchese — APPROVED** [review]

**Review by cblecker — APPROVED** [review]

**Pull request merged**